src/web-ui/src/public/components/AmplifyMap.vue

```diff
@@ -0,0 +1,259 @@
+<template>
+  <div id="mapContainer" style="height: 100%; width: 100%"></div>
+</template>
+
+<script>
+import mapbox from 'mapbox-gl';
+import "mapbox-gl/dist/mapbox-gl.css";
+import {Credentials, Signer} from "@aws-amplify/core";
+import Location from "@/location/Location";
+
+const resourceName = process.env.VUE_APP_LOCATION_RESOURCE_NAME;
+const locationApi = new Location();
+
+export default {
+  name: "AmplifyMap",
+  props: {
+    devicePositions: {
+      type: Array,
+      default: () => []
+    },
+    routes: {
+      type: Array,
+      default: () => []
+    },
+    storePosition: {
+      type: Array,
+      default: () => [0, 0]
+    },
+    zoom: {
+      type: Number,
+      default: () => 14
+    },
+  },
+  data () {
+    return {
+      map: null,
+      mapLoaded: false
+    }
+  },
+  mounted () {
+    this.getCreds()
+  },
+  methods: {
+    getCreds() {
+      Credentials.get()
+        .then(creds => this.creds = creds)
+        .then(this.loadMap)
+    },
+    async loadMap() {
+      if (!this.storePosition) {
+        return
+      }
+      this.map = new mapbox.Map({
+        attributionControl: false,
+        center: this.storePosition || [0, 0],
+        container: "mapContainer",
```

Comment:
The API can also take an `HTMLElement` so using a `ref` would be more preferable here.